test_spack.py

```diff
@@ -0,0 +1,432 @@
+
+
+class EccodesTest(unittest.TestCase):
+    package_name = 'eccodes'
+    depends_on = {}
+    machines = all_machines
+
+
+class EckitTest(unittest.TestCase):
+    package_name = 'eckit'
+    depends_on = {'ecbuild'}
+    machines = all_machines
+
+
+class GridToolsTest(unittest.TestCase):
+    package_name = 'gridtools'
+    depends_on = {'cuda'}
+    machines = all_machines
+
+
+class IconTest(unittest.TestCase):
+    package_name = 'icon'
+    depends_on = {'serialbox', 'eccodes', 'claw'}
+    machines = all_machines
+
+    # def test_install(self):
+    #     # TODO: Decide if we want to integrate this test or not. It has been used lately here: From https://github.com/C2SM/spack-c2sm/pull/289
+    #     run('spack install icon@nwp%pgi icon_target=gpu +claw')
+
+    # TODO: Reactivate once the test works!
+    # def test_devbuild_cpu(self):
+    #     # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+    #     run('git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git')
+    #     run('mkdir -p icon-cscs/pgi_cpu')
+    #     run('touch a_fake_file.f90', cwd='icon-cscs/pgi_cpu')
+        
+    #     try:
+    #         run('spack dev-build -i -u build icon@dev-build%pgi config_dir=./.. icon_target=cpu', cwd='icon-cscs/pgi_cpu')
+    #     finally:
+    #         run('rm -rf icon-cscs')
+
+    # TODO: Reactivate once the test works!
+    # def test_devbuild_gpu(self):
+    #     # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+    #     run('git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git')
+    #     run('mkdir -p icon-cscs/pgi_gpu')
+    #     run('touch a_fake_file.f90', cwd='icon-cscs/pgi_gpu')
+
+    #     try:
+    #         run('spack dev-build -i -u build icon@dev-build%pgi config_dir=./.. icon_target=gpu', cwd='icon-cscs/pgi_gpu')
+    #     finally:
+    #         run('rm -rf icon-cscs')
+
+
+class Int2lmTest(unittest.TestCase):
+    package_name = 'int2lm'
+    depends_on = {'cosmo-grib-api-definitions', 'cosmo-eccodes-definitions', 'libgrib1'}
+    machines = {'tsa'}
+
+    def test_install(self):
+        # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+        run('spack install int2lm@c2sm_master%pgi')
+
+    # def test_install_test(self):
+    #     # TODO: Decide if we want to integrate this test or not. It has been used lately here: From https://github.com/C2SM/spack-c2sm/pull/319
+    #     run('spack install --test=root int2lm@c2sm_master%gcc')
+
+    def test_install_no_pollen(self):
+        # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+        run('spack install int2lm@org_master%pgi pollen=False')
+
+
+class IconDuskE2ETest(unittest.TestCase):
+    package_name = 'icondusk-e2e'
+    depends_on = {'atlas_utilities', 'dawn4py', 'dusk', 'atlas', 'cuda'}
+    machines = all_machines
+
+
+class IconToolsTest(unittest.TestCase):
```

Comment:
For the Icontools I would like to test the following setup:
run('spack install --test=root icontools@c2sm-master%cce')
The problem is that cce only runs on Daint, whereas for Tsa we rely on gcc.